kernel/base/src/main/java/com/twosigma/beakerx/util/DateUtil.java

```diff
@@ -0,0 +1,30 @@
+package com.twosigma.beakerx.util;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+
+public final class DateUtil {
+
+  public static long dateToLong(Object datelike) {
```

Comment:
could overloading be used here to avoiding going back to Object?

Reply:
It would result in more code duplication because we'd have to do the `instanceof` checks in multiple places to cast the input to the appropriate type to pass into the overloaded functions